Cabal/Cabal.sublime-syntax

```diff
@@ -0,0 +1,91 @@
+%YAML 1.2
+---
+# Derived from https://github.com/JustusAdam/language-haskell/blob/2859b32113407977f0edcb6c19f0fe3eaae1294f/syntaxes/cabal.tmLanguage
+name: Cabal
+file_extensions:
+  - cabal
+scope: source.cabal
+contexts:
+  main:
+    - match: |-
+        (?ix)(\n|^)
+            ( name
+            | version
+            | cabal-version
+            | build-type
+            | license(-file)?
+            | copyright
+            | author
+            | maintainer
+            | stability
+            | homepage
+            | bug-reports
+            | package-url
+            | synopsis
+            | data-(files|dir)
+            | description
+            | category
+            | extra-(source|doc|tmp)-files
+            | tested-with
+            ):
+      scope: keyword.other.cabal
+    - match: |-
+        (?ix)^[ \t]+
+            ( build-depends
+            | other-modules
+            | hs-source-dirs
+            | ((other|default)-)?extensions
+            | build-tools
+            | buildable
+            | ghc(-(prof|shared))?-options
+            | (install-)?includes
+            | include-dirs
+            | (c|js)-sources
+            | extra(-ghci)?-libraries
+            | extra-lib-dirs
+            | (cc|cpp|ld)-options
+            | pkgconfig-depends
+            | default-language
+            | frameworks
+            | default
+            | manual
+            | location
+            | branch
+            | tag
+            | subdir
+            | exposed(-modules)?
+            | reexported-modules
+            | main-is
+            | type
+            | test-module
+            | description
+            | setup-depends
+            | mixins
+            ):
+      scope: keyword.other.cabal
+    - match: (==|>=|<=|<|>|\|\||&&|!)
+      scope: keyword.operator.cabal
+    - match: '(?<=[^\w])\d+(\.\d+)*(\.\*)?'
+      scope: constant.numeric.cabal
+    - match: '\w+:/(/[\w._\-\d%])+(\?[\w.+_\-\d%]+)(&[\w._+\-\d%]+)*'
+      scope: markup.underline.link.cabal
+    - match: |-
+        ^(?ix:
+        ( library
+        | custom-setup
+        ))$
+      scope: entity.name.section.cabal
+    - match: |-
+        ^(?ix:
+        ( executable
+        | flag
+        | test-suite
+        | source-repository
+        ))( |\t)+([\w\-_]+)$
+      captures:
+        1: entity.name.section.cabal
+        3: entity.name.function.cabal
+    - match: '^[ \t]*(if|else)'
+      scope: keyword.control.cabal
+    - match: ^\s*--.*$
```

Comment:
You should implement line comments with a context push to allow with_prototype matches within them, in case this ever comes up. You should also include the trailing newline in the match.